Detect correct file coding

thom-sch opened the pull request:
This commit relates to issue #43 
An enhancement in collect.py tries to guess the encoding of the input file to avoid errors.
Originally they are caused on trying to convert the input file to UTF8 encoding in lines 160..169:
If the text.encode / text.decode fails (due to wrong file encoding) this is not really recognized ("pass" in the except statement). Variable "text" may have the wrong type (string vs. bytearray) and "text.encode('utf8')" in line 169 fails.

avkonst commented:
Thanks. Very scientific file reader. I have approved CI to run. If tests pass, I will merge.

avkonst commented:
some parser tests are failing. please try to run locally and fix the tests. You may regenerate gold files which seems like failing the tests.

thom-sch commented:
I've fixed the errors - mainly caused by wrong UTF8 character ranges; added also some support for Python 2.7 which causes the Python 2.7 tests to fail. No need to regenerate the gold files!
Tests successfully on my machine:
- Win10
- Python 2.7
- Python 3.7
- Python 3.9
- Python 3.10

Regards from Meerbusch
Thomas
